src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherState.scala

```diff
@@ -88,6 +90,9 @@ case class BlockFetcherState(
         .ensure("Given headers are not sequence with already fetched ones")(_.head.number <= nextToLastBlock)
         .ensure("Given headers aren't better than already fetched ones")(_.last.number > lastBlock)
         .ensure("Given headers should form a sequence without gaps")(HeadersSeq.areChain)
+        .ensure("Given headers do not form a chain with already stored ones")(headers =>
+          (waitingHeaders.lastOption, headers.headOption).mapN(_ isParentOf _).getOrElse(true)
```

Comment:
I was thinking... should we maybe check this condition to see if there's any way of concatenating both branches? This condition might be false only because we received a header from a NewBlock

Reply:
Given [this](https://github.com/input-output-hk/mantis/pull/787#discussion_r520688749) conversation, I think this comment doesn't apply